### PR TITLE
No longer require CASA for coordsys to WCS conversion

### PR DESCRIPTION
**Pull request by astrofrog**
This is a work in progress which makes it so that we no longer need CASA to convert the CASA coordsys to Astropy WCS. Remaining to-dos:

* [x] Add changelog entry
* [x] Try and add some more test cases
* [x] Add support for the 'linear' axis type, which is relevant if no WCS is set
* [x] Check casacore code for other supported spectral systems (there are a few in ``FITSCoordinateUtil.cc``)
* [x] Determine why CDELT disagrees at the 1e-5 relative level for spectral axis (see #614)
* [x] Rebase once https://github.com/radio-astro-tools/spectral-cube/pull/609 is merged

With this PR it's now possible to fully load CASA spectral cubes without CASA!

### CODE REVIEW

**Comment by astrofrog** [comment]
@e-koch - could you try out this PR? Note that you could try it in an environment without CASA since CASA should no longer be required to read in cubes with ``SpectralCube.read``

**Comment by astrofrog** [comment]
(note that this is not worth reviewing line by line until #609 is merged, since this PR includes the changes from there)

**Comment by e-koch** [comment]
@astrofrog - Everything seems to be working using the same cubes I used testing #609!

**Comment by astrofrog** [comment]
@e-koch - great!

File: spectral_cube/io/casa_wcs.py [hunk 1]
```diff
@@ -1,11 +1,57 @@
...
+}
+
+for system in EQUATORIAL_SYSTEMS:
+    AXES_TO_CTYPE[system] = {'Right Ascension': 'RA--',
+                             'Declination': 'DEC-'}
+
+
+def sanitize_unit(unit):
+    if unit == "'":
+        return 'arcmin'
```
Comment (keflavich) [inline spectral_cube/io/casa_wcs.py]: do we need `"` for arcsec, or does that not appear?

File: spectral_cube/io/casa_wcs.py [hunk 2]
```diff
@@ -14,24 +60,142 @@ def wcs_casa2astropy(ndim, coordsys):
...
+            worldmap[index] = value
+
+    # Now iterate through the different coordinate types to populate the WCS
+
+    header['WCSAXES'] = np.max([np.max(idx) + 1 for idx in worldmap.values()])
+
+    # Initialize PC
+    for i in range(header['WCSAXES']):
+        for j in range(header['WCSAXES']):
+            header[f'PC{i+1}_{j+1}'] = 0.
```
Comment (keflavich) [inline spectral_cube/io/casa_wcs.py]:
```suggestion
    for ii in range(header['WCSAXES']):
        for jj in range(header['WCSAXES']):
            header[f'PC{ii+1}_{jj+1}'] = 0.
```

File: spectral_cube/io/casa_wcs.py [hunk 3]
```diff
@@ -14,24 +60,142 @@ def wcs_casa2astropy(ndim, coordsys):
...
+
+    for coord_type in ('direction', 'spectral', 'stokes', 'linear'):
+
+        for index in range(len(worldmap)):
+            if f'{coord_type}{index}' in coordsys:
+                break
+        else:
+            continue
```
Comment (keflavich) [inline spectral_cube/io/casa_wcs.py]: Just commenting that I have never before seen the `for/else` used in practice.    It's uncommon enough that I would suggest including a comment about it; at first I was convinced this was a typo.

File: spectral_cube/io/casa_wcs.py [hunk 4]
```diff
@@ -14,24 +60,142 @@ def wcs_casa2astropy(ndim, coordsys):
...
+                header[f'PV{idx2}_2'] = 0.
+            header[f'PC{idx1}_{idx1}'] = data['pc'][0, 0]
+            header[f'PC{idx1}_{idx2}'] = data['pc'][0, 1]
+            header[f'PC{idx2}_{idx1}'] = data['pc'][1, 0]
+            header[f'PC{idx2}_{idx2}'] = data['pc'][1, 1]
+        elif coord_type == 'stokes':
+            idx = worldmap[index][0] + 1
+            header[f'CTYPE{idx}'] = 'STOKES'
+            header[f'CRVAL{idx}'] = data['crval'][0]
+            header[f'CRPIX{idx}'] = data['crpix'][0] + 1
```
Comment (keflavich) [inline spectral_cube/io/casa_wcs.py]: so... this means CASA stores CRPIX in zero-indexed, not one-indexed form?

Comment (keflavich) [reply spectral_cube/io/casa_wcs.py]: (this comment applies elsewhere too - I'm just observing, I don't think any code change or even commentary is needed)

File: spectral_cube/io/casa_wcs.py (hunk 1 above)
Comment (astrofrog) [reply spectral_cube/io/casa_wcs.py]: I haven't managed to find files that use this, but no harm in adding it, so I've done this.

File: spectral_cube/io/casa_wcs.py (hunk 3 above)
Comment (astrofrog) [reply spectral_cube/io/casa_wcs.py]: I've added a comment. The ``else`` for a ``for`` loop is quite useful for situations like this!

File: spectral_cube/io/casa_wcs.py (hunk 4 above)
Comment (astrofrog) [reply spectral_cube/io/casa_wcs.py]: Yes it seems so, which would make sense internally for calculations.

**Pull request merged**